project-icp/bee-pollinator-app
# Recolor CDL and Update CDL-Related CSS

# Overview
We want to apply the scheme of the original CDL colormap to the new reclassified one, but the original has some problems:
- Some crops that belong to different groups have the same color
- Some crops are an ambiguous, nasty gray that doesn't fit in with the color scheme and makes the crop look like it's "no data"

I've added a script `recolor_cdl.py` that tries to make a new colormap for each of the 47 crop groups "inspired" by the original. It 
1. removes all the gray and duplicate values from the original colormap
2. adds the not-duplicate/not-gray values using the reclass map to the new colormap by the grouped crop id
3. Some groups/keys won't have a value in the new colormap, so it uses whatever is leftover/unused from the old, not-duplicate/not-gray one to fill in the gaps

# Demo
### The raw, original CDL
![screen shot 2016-12-30 at 1 17 03 pm](https://cloud.githubusercontent.com/assets/7633670/21570522/bb20e746-ce94-11e6-9519-8249325ddb70.png)

### The new recolored and reclassified CDL
![screen shot 2016-12-30 at 1 15 58 pm](https://cloud.githubusercontent.com/assets/7633670/21570597/78bd4f38-ce95-11e6-94c7-c246b5393922.png)

### Now with 78% more pastels
<img width="602" alt="screen shot 2016-12-30 at 1 09 34 pm" src="https://cloud.githubusercontent.com/assets/7633670/21570529/c14a69ee-ce94-11e6-92e8-0e1a2ab03aee.png">

# Testing Instructions
Assuming you have the two CDL tifs in your `/opt/icp-crop-data` dir, pull this branch

1. The recolor script
```
vagrant ssh worker
cd /vagrant/src/icp/pollinator/src/pollinator/reclass/
python recolor_cdl.py
```
  - When it's done you'll have `./cdl_recolor.tif`
   -   Inspect it in QGIS
   -   Temporarily replace `/opt/icp-crop-data/cdl_reclass_lzw_5070.tif` with it and confirm the model still runs

2. The app
`./scripts/bundle.sh`
- Draw an AOI 
- The landcover colors should match the new colormap's (`berries id=20, almonds=33, buckwheat=12, melons=18`) 
- The colors in the chart should match the land cover's

Connects #95
